toolchain/check/testdata/class/no_definition_in_impl_file.carbon

```diff
@@ -0,0 +1,118 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+// TIP: To test this file alone, run:
+// TIP:   bazel test //toolchain/testing:file_test --test_arg=--file_tests=toolchain/check/testdata/class/no_definition_in_impl_file.carbon
+// TIP: To dump output, run:
+// TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/check/testdata/class/no_definition_in_impl_file.carbon
+
+// --- a.carbon
+
+library "a";
+
+class AForwardDecl;
+
+// --- a.impl.carbon
+
+impl library "a";
+
+class AForwardDecl;
+
+class AForwardDecl {}
+
+// --- b.carbon
```

Comment:
What do you think of adding two more test cases:

1. The API file has no forward declaration, the impl file has a forward declaration.
2. The API file has a forward declaration, the impl file has nothing.

I think both of these would be handled correctly; I'm only thinking about test coverage.